test/execution/ast_type_test.cpp

```diff
@@ -19,6 +19,27 @@ class TypeTest : public TplTest {
 
   ast::Identifier Name(const std::string &s) { return Ctx()->GetIdentifier(s); }
 
+  ast::Identifier PaddingName(uint32_t offset) { return Ctx()->GetIdentifier("__pad$" + std::to_string(offset) + "$"); }
+
+  void CheckArrayType(StructType *struct_type, ast::Identifier name, uint32_t offset, uint32_t length,
```

Comment:
This interface seems a little unintuitive w.r.t. what it does. Types don't have names, and the name can be derived from the offset, so isn't required in the interface. But, at a higher level, I think what you want is:

```
/** Checks if the element at the given offset is padding of the given size. */
void CheckIsPadding(StructType *type, std::size_t offset, std::size_t expected_padding_size);
```

The implementation pulls out the field by name (using the name derived from the offset) and (1) checks it exists and (2) checks the type is indeed an array of the given size and whose elements are bytes using an another `void CheckIsArrayType(Type *type, std::size_t num_elems, Type *elem_type);`